assets/source/setup-guide/app/steps/SetupTracking.js

```diff
@@ -283,14 +286,40 @@ const SetupTracking = ( { view = 'settings' } ) => {
 	};
 
 	const handleCompleteSetup = async () => {
-		recordEvent( 'pfw_setup', {
-			target: 'complete',
-			trigger: 'setup-tracking-complete',
-		} );
-		// Force reload WC admin page to initiate the relevant dependencies of the Dashboard page.
-		const path = getNewPath( {}, '/pinterest/settings', {} );
-
-		window.location = new URL( wcSettings.adminUrl + path );
+		try {
+			const result = await connectAdvertiser(
+				appSettings.tracking_advertiser,
+				appSettings.tracking_tag
+			);
```

Comment:
`connectAdvertiser` returns boolean true/false. If it returns false (which seems to mean account is not connected), it should not proceed with `recordEvent` and redirect with `window.location`, right? 🤔

Reply:
The boolean true/false indicates if the account was reconnected (reconnected means that a new advertiser or tag is connected to the platform). If the advertiser and tag were already connected (no changes in the previous values), `connectAdvertiser` will return false.
If there is a problem connecting the advertiser the method will throw an error that will be handled by the catch block.